samples/transient-clients/Core_6/Publisher/Program.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Threading.Tasks;
+using NServiceBus;
+
+static class Program
+{
+    static void Main()
+    {
+        AsyncMain().GetAwaiter().GetResult();
+    }
+
+    static async Task AsyncMain()
+    {
+        Console.Title = "Samples.TransientClients.Publisher";
+        var endpointConfiguration = new EndpointConfiguration("Samples.TransientClients.Publisher");
+        endpointConfiguration.UseSerialization<JsonSerializer>();
+        endpointConfiguration.UsePersistence<InMemoryPersistence>();
+        endpointConfiguration.UseTransport<MsmqTransport>();
+
+        endpointConfiguration.SendFailedMessagesTo("error");
+
+        endpointConfiguration.Conventions().DefiningEventsAs(t => t == typeof(StockTick));
```

Comment:
This allows the Client (non-NSB) to share the message type with the NServiceBus endpoints.